### PR TITLE
feat: Restructure groups file structures and add checks on edit routes.

### LINKED ISSUE #7626: Groups: Refactor group file structure and add permissions
- [ ] Currently file structure of group needs refactoring.
- [ ] Restrict other user to access groups.

### PR DESCRIPTION
**Pull request by sachinchauhan2889**
Fixes #7626 

- [X] Restructure groups file structure
- [X] Add permissions on edit routes.

### New file structures

- **My-groups**
> - list
> - following

- **groups**
> - create
> - team
> - edit
> > - settings
> > - followers
> > - events

- **group-public**

### CODE REVIEW

File: app/router.js [hunk 1]
```diff
@@ -197,6 +194,11 @@ Router.map(function() {
   this.route('my-sessions', function() {
     this.route('list', { path: '/:session_status' });
   });
+  this.route('my-groups', function() {
+    this.route('list');
+    this.route('following');
+  });
+  this.route('group-public', { path: 'g/:group_id' });
```
Comment (maze-runnar) [inline app/router.js:201]:
```suggestion
  this.route('group-public', { path: '/g/:group_id' });
```

File: app/routes/groups/edit/events.js [hunk 2]
```diff
@@ -35,4 +35,10 @@ export default class EventsRoute extends Route.extend(AuthenticatedRouteMixin) {
       })
     });
   }
+
+  afterModel(model) {
+    if (this.authManager.currentUser.email !== model.group.user.get('email')) {
```
Comment (maze-runnar) [inline app/routes/groups/edit/events.js:40]: what if user is admin?

File: app/routes/groups/edit/settings.js [hunk 3]
```diff
@@ -17,4 +17,10 @@ export default class SettingsRoute extends Route.extend(AuthenticatedRouteMixin)
       })
     });
   }
+
+  afterModel(model) {
+    if (this.authManager.currentUser.email !== model.group.user.get('email')) {
```
Comment (maze-runnar) [inline app/routes/groups/edit/settings.js:22]: here too.

**Comment by sachinchauhan2889** [comment]
> When I am going into the group followers section and then clicking on the event tab or settings tab, it is redirecting me to the index page.

Not able to reproduce. @maze-runnar

**Comment by maze-runnar** [comment]
> > When I am going into the group followers section and then clicking on the event tab or settings tab, it is redirecting me to the index page.
> 
> Not able to reproduce. @maze-runnar

Have you checked on deployment url?

**Comment by MeghalBisht** [comment]
> When I am going into the group followers section and then clicking on the event tab or settings tab, it is redirecting me to the index page.

Not able to reproduce either.

**Review by MeghalBisht — APPROVED** [review]

**Comment by sachinchauhan2889** [comment]
> When I am going into the group followers section and then clicking on the event tab or settings tab, it is redirecting me to the index page.

Non admin.

**Review by ranjitkshah — APPROVED** [review]

**Pull request merged**